git/repo.go

```diff
@@ -21,7 +21,7 @@ const (
 var (
 	ErrNoChanges = errors.New("no changes made in repo")
 	ErrNotReady  = errors.New("git repo not ready")
-	ErrNoConfig  = errors.New("git repo has not valid config")
+	ErrNoConfig  = errors.New("git repo does not have valid config")
```

Comment:
Should this be used somewhere? Since no code references this value, this PR doesn't
> Let fluxd answer queries without a git repo supplied,

so much as it does
>Let fluxd answer queries before the supplied git repo is ready,

right?

Reply:
> Since no code references this value

https://github.com/weaveworks/flux/blob/master/git/repo.go#L156

but .. I take your general point, which is that this case isn't handled (or isn't handled well). Possibly there should be another ReadOnlyReason for "No git repo supplied", to distinguish it from "git repo isn't ready".

Reply:
Oh I see, that state is not reachable, because the git.Repo is initialised in the state RepoNew and there are no transitions into RepoNoConfig.

OK, I have fixed that (most recent commit). If the git.Repo is given an empty origin URL, it is initialised in RepoNoConfig. This percolates out to the API, where it's represented as `const ReadOnlyNoRepo ReadOnlyReason = "NoRepo"`.

Reply:
> there are no transitions into RepoNoConfig

I worded my comment poorly enough that it wasn't even obvious to me this is what I actually meant... Yep, there was no code which _assigned_ that value to anything.

If you're going to link code you should use [the permalink](https://github.com/weaveworks/flux/blob/1a860b16de3eb86f4a17c088a805da61c7bc8e4c/git/repo.go#L156). You never know how much time you might save the next code archeologist 😄.

Reply:
> If you're going to link code you should use the permalink

<kbd>y</kbd> is a nice shortcut for this when looking at a branch HEAD